aiida/cmdline/commands/cmd_export.py

```diff
@@ -212,5 +227,12 @@ def migrate(input_file, output_file, force, silent, archive_format, version):
             with tarfile.open(output_file, 'w:gz', format=tarfile.PAX_FORMAT, dereference=True) as archive:
                 archive.add(folder.abspath, arcname='')
 
+        if in_place:
+            os.rename(output_file, input_file)
+            tempdir.cleanup()
+
         if not silent:
-            echo.echo_success('migrated the archive from version {} to {}'.format(old_version, new_version))
+            if new_version > old_version:
+                echo.echo_success('migrated the archive from version {} to {}'.format(old_version, new_version))
+            else:
+                echo.echo_success('archive was already at latest version {}'.format(new_version))
```

Comment:
I think I would move this check to the beginning of the command. With the `Archive` context manager util class, you can relatively easily get the version number and if the same you can skip all the unpackking/migrating/repacking logic. By checking the version number, the archive will have to be unpacked first, so that cost cannot be skipped but at least we are not repacking it again for nothing. Once we have the new export format that should even make the introspection cheap, even the unnecessary unpacking will no longer be necessary

Reply:
Right. It does change the user interface a bit: this change means you can no longer count on the new archive being present in the `outfile` in all cases (unless you use `--in-place`).
I agree however that we may want to save the repackaging step...

Reply:
Change made!